src/Compilers/CSharp/Test/Symbol/Symbols/Source/NullablePublicAPITests.cs

```diff
@@ -0,0 +1,634 @@
+                },
+                compilation =>
+                {
+                    var c = compilation.GetTypeByMetadataName("C");
+                    return c.GetMembers().OfType<IEventSymbol>().ToArray();
+                },
+                member => member.NullableAnnotation,
+                testMetadata: true,
+                PublicNullableAnnotation.NotAnnotated,
+                PublicNullableAnnotation.Annotated,
+                PublicNullableAnnotation.Disabled,
+                PublicNullableAnnotation.Annotated);
+        }
+
+        private void VerifyAcrossCompilations<T>(string source,
+                                                 DiagnosticDescription[] nullableEnabledErrors,
+                                                 DiagnosticDescription[] nullableDisabledErrors,
+                                                 Func<CSharpCompilation, T[]> memberFunc,
+                                                 Func<T, PublicNullableAnnotation> nullabilityFunc,
+                                                 bool testMetadata,
+                                                 params PublicNullableAnnotation[] expectedNullabilities)
+        {
+
+            var comp1 = CreateCompilation(source, options: WithNonNullTypesTrue());
+            comp1.VerifyDiagnostics(nullableEnabledErrors);
+            verifyCompilation(comp1);
+
+            var comp2 = CreateCompilation(source, options: WithNonNullTypesFalse());
+            comp2.VerifyDiagnostics(nullableEnabledErrors);
+            verifyCompilation(comp2);
+
+            var comp3 = CreateCompilation(source, parseOptions: TestOptions.Regular7_3, skipUsesIsNullable: true);
+            comp3.VerifyDiagnostics(nullableDisabledErrors);
+            verifyCompilation(comp3);
+
+            if (!testMetadata) return;
+
+            var comp1Emit = comp1.EmitToImageReference();
+            var comp4 = CreateCompilation("", references: new[] { comp1Emit }, options: WithNonNullTypesTrue());
+            comp4.VerifyDiagnostics();
+            verifyCompilation(comp4);
+
+            var comp2Emit = comp2.EmitToImageReference();
+            var comp5 = CreateCompilation("", references: new[] { comp2Emit }, options: WithNonNullTypesFalse());
+            comp5.VerifyDiagnostics();
+            verifyCompilation(comp5);
+
+            var comp6 = CreateCompilation("", references: new[] { comp1Emit }, parseOptions: TestOptions.Regular7_3);
+            comp6.VerifyDiagnostics();
+            verifyCompilation(comp6);
+
+            void verifyCompilation(CSharpCompilation compilation)
+            {
+                var members = memberFunc(compilation);
+
+                Assert.Equal(members.Length, expectedNullabilities.Length);
+
+                for (int i = 0; i < expectedNullabilities.Length; i++)
+                {
+                    Assert.Equal(expectedNullabilities[i], nullabilityFunc(members[i]));
+                }
```

Comment:
`AssertEx.Equal(members.Select(nullabilityFunc), expectedNullabilities);` will report a diff of the members. And an `itemInspector:` arg can format the result if needed.